pkg/trace/pb/span_gen.go

```diff
@@ -1,357 +1,290 @@
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.ParentID = 0
+				break
+			}
+			z.ParentID, bts, err = parseUint64Bytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "ParentID")
 				return
 			}
 		case "start":
-			if dc.IsNil() {
-				z.Start, err = 0, dc.ReadNil()
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.Start = 0
 				break
 			}
-
-			z.Start, err = parseInt64(dc)
+			z.Start, bts, err = parseInt64Bytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "Start")
 				return
 			}
 		case "duration":
-			if dc.IsNil() {
-				z.Duration, err = 0, dc.ReadNil()
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.Duration = 0
 				break
 			}
-
-			z.Duration, err = parseInt64(dc)
+			z.Duration, bts, err = parseInt64Bytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "Duration")
 				return
 			}
 		case "error":
-			if dc.IsNil() {
-				z.Error, err = 0, dc.ReadNil()
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.Error = 0
 				break
 			}
-
-			z.Error, err = parseInt32(dc)
+			z.Error, bts, err = parseInt32Bytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "Error")
 				return
 			}
 		case "meta":
-			if dc.IsNil() {
-				z.Meta, err = nil, dc.ReadNil()
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.Meta = nil
 				break
 			}
-
-			var zwht uint32
-			zwht, err = dc.ReadMapHeader()
+			var zb0002 uint32
+			zb0002, bts, err = msgp.ReadMapHeaderBytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "Meta")
 				return
 			}
-			if z.Meta == nil && zwht > 0 {
-				z.Meta = make(map[string]string, zwht)
+			if z.Meta == nil && zb0002 > 0 {
+				z.Meta = make(map[string]string, zb0002)
 			} else if len(z.Meta) > 0 {
 				for key := range z.Meta {
 					delete(z.Meta, key)
 				}
 			}
-			for zwht > 0 {
-				zwht--
-				var zxvk string
-				var zbzg string
-				zxvk, err = parseString(dc)
+			for zb0002 > 0 {
+				var za0001 string
+				var za0002 string
+				zb0002--
+				za0001, bts, err = parseStringBytes(bts)
 				if err != nil {
+					err = msgp.WrapError(err, "Meta")
 					return
 				}
-				zbzg, err = parseString(dc)
+				za0002, bts, err = parseStringBytes(bts)
 				if err != nil {
+					err = msgp.WrapError(err, "Meta", za0001)
 					return
 				}
-				z.Meta[zxvk] = zbzg
+				z.Meta[za0001] = za0002
 			}
 		case "metrics":
-			if dc.IsNil() {
-				z.Metrics, err = nil, dc.ReadNil()
+			if msgp.IsNil(bts) {
+				bts, err = msgp.ReadNilBytes(bts)
+				z.Metrics = nil
 				break
 			}
-
-			var zhct uint32
-			zhct, err = dc.ReadMapHeader()
+			var zb0003 uint32
+			zb0003, bts, err = msgp.ReadMapHeaderBytes(bts)
 			if err != nil {
+				err = msgp.WrapError(err, "Metrics")
 				return
 			}
-			if z.Metrics == nil && zhct > 0 {
-				z.Metrics = make(map[string]float64, zhct)
+			if z.Metrics == nil && zb0003 > 0 {
+				z.Metrics = make(map[string]float64, zb0003)
```

Comment:
len(z.Metrics) > 0 below can happen when a field is sent multiple times?

Reply:
That or if the a message is deserialized in an object that has already a non empty Metrics field.